climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1387 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define WildFire class.
+"""
+
+__all__ = ['WildFire']
+
+import logging
+import warnings
+import itertools
+from datetime import date
+from functools import partial
+
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from scipy import sparse
+from sklearn.neighbors import BallTree
+from sklearn.cluster import DBSCAN
+import numba
+import geopandas as gpd
+from shapely.geometry import Point
+from shapely.ops import transform
+import pyproj
+
+from climada.hazard.centroids.centr import Centroids
+from climada.hazard.base import Hazard
+from climada.hazard.tag import Tag as TagHazard
+from climada.util.constants import ONE_LAT_KM, DEF_CRS
+import climada.util.dates_times as u_dt
+import climada.util.alpha_shape as u_alpha
+warnings.simplefilter(action='ignore', category=FutureWarning)
+
+LOGGER = logging.getLogger(__name__)
+
+HAZ_TYPE = 'WF'
+""" Hazard type acronym for Wild Fire, might be changed to WFseason or WFsingle """
+
+CLEAN_THRESH = 30
+""" Minimal confidence value for the data from MODIS instrument to be use as input"""
+
+RES_DATA = 1.0
+""" Resolution of the data if no data origin provided (km) """
+
+BLURR_STEPS = 4
+""" steps with exponential decay for fire propagation matrix """
+
+class WildFire(Hazard):
+
+    """Contains wild fire events.
+
+    Wildfires comprise the challenge that the definition of an event is unclear.
+    Reporting standards vary accross regions and over time. Hence, to have
+    consistency, we consider an event as a whole fire season. This allows
+    consistent risk assessment across the global and over time. Events that
+    refer to a fire season have the tag 'WFseason'.
+
+    In order to perform concrete case studies or calibrate impact functions,
+    events can be displayed as single fires. In that case they have the tag
+    'WFsingle'.
```

Comment:
Please a note to explain when can one use single fires as events?
```suggestion
    In order to perform concrete case studies or calibrate impact functions,
    single fires can also be used instead for events when the data is available.
    In that case, the Hazard has the tag 'WFsingle'.
```

Reply:
I think that's explained ("In order to perform concrete case studies or calibrate impact functions"

Reply:
That is not when, but what for.

Reply:
I don't get it what you mean by that?

Reply:
After rereading I think it is fine. Thanks!